src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.Comparison.cs

```diff
@@ -0,0 +1,412 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Text;
+using System.Text.Unicode;
+using System.Numerics;
+
+#pragma warning disable SA1121 // Use built-in type alias
+using SkipChecks = System.Boolean;
+using CheckBytes = System.Byte;
+using CheckChars = System.Char;
+using CheckNonAscii = System.Byte;
```

Comment:
I don't understand the purpose of this. Especially with multiple names mapped to the same type.

Reply:
> I don't understand the purpose of this.

The private `Equals` method compares `ROS<char>` with `ROS<byte>`:

```cs
Equals<TCheck>(ReadOnlySpan<char> chars, ReadOnlySpan<byte> bytes) where TCheck : struct
```

is used by the following public methods:

```cs
public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
public static unsafe bool StartsWith(ReadOnlySpan<byte> text, ReadOnlySpan<char> value)
public static unsafe bool EndsWith(ReadOnlySpan<byte> text, ReadOnlySpan<char> value)
public static unsafe bool StartsWith(ReadOnlySpan<char> text, ReadOnlySpan<byte> value)
public static unsafe bool EndsWith(ReadOnlySpan<char> text, ReadOnlySpan<byte> value)
```

`Equals` should not throw when any of the provided arguments contain non-ascii chars.
But `StartsWith` and `EndsWith` should throw only when `value` contains non-ascii chars. The thing is that value can be both `char` and `byte`. My goal of using the aliases was not express "in this case bytes are the value, so check the bytes" (same for chars) and "don't perform any checks". 

> Especially with multiple names mapped to the same type.

That is true. I will try to come up with simpler solution.